### ISSUE TITLE
isAuthorized returning true if Auth allow is set

### ISSUE THREAD

**Issue by cercos** [CakeDC/users#527]
The isAuthorized() and link() method on the AuthLinkComponent returns true if the action of the link matches an allowed action on the controller for the current request.  i.e.

FirstController for the mock request
```
class FirstController extends AppController
{
    public function beforeFilter(Event $event)
    {
        $this->Auth->allow(['index');
    }
}
```

A link that has permissions set
`$this->User->AuthLink->link(''I'm a link', ['controller' => 'SecondController', 'action' => 'index'])`

If that link is placed on an action inside FirstController it will return true even though they aren't authorized for the url on the SecondController.

**Comment by steinkel** [comment]
Could you please submit a failing unit test or the complete code example to reproduce this issue? I was not able to do it in my test environment.

Thanks!

**Comment by cercos** [comment]
Ive made a [test app](https://github.com/cercos/test_auth) that shows what I mean.  Because the PagesController has the secret action allowed the AuthLink helper returns true for the link even though it's not authorized.

**Comment by steinkel** [comment]
Interesting, I'll check this ASAP, thank you for the app

**Comment by cercos** [comment]
I believe the issue is with the _isActionAllowed() method on the UsersAuthComponent.  It's checking if the link is an allowed action on the current controller the component is attached to not the controller in the url.

**Comment by steinkel** [comment]
Ah I see what you mean here, you are trying to check permissions on an action from another controller. In this case, we would need to check if the action is in the allowed set of actions for the 2nd controller, which would involve creating an instance of the controller and executing initialize and beforeFilter. Not only this could be a performance issue, but also it could trigger additional business logic, associated to these 2 methods. How to fix this use case?
* Let _isActionAllowed return false if the controller name != current controller and debug this to debug.log, and fix the documentation to warn the developer

Additionally, if you really need this feature in your project, I could think of a configuration param `Users.Controller.allowedActions` that will have the array of `'ControllerName' => ['allowedAction']` and a method in AppController beforeFilter to apply this configuration to AuthComponent. If we use this configuration we could avoid the 2nd instance generation and all the side effects.

What do you think?

**Comment by cercos** [comment]
Good idea if I understand you correctly.

**Comment by steinkel** [comment]
closed per  #540
